github/compliance/model.py

```diff
@@ -174,8 +191,11 @@ def result_groups(self) -> tuple[ScanResultGroup]:
             return ()
 
         for result in self.results:
-            artifact = cnudie.iter.artifact_from_node(result.scanned_element)
-            group_name = f'{result.scanned_element.component.name}:{artifact.name}'
+            if isinstance(result.scanned_element, (cnudie.iter.ResourceNode, cnudie.iter.SourceNode)): # noqa: E501
+                artifact = cnudie.iter.artifact_from_node(result.scanned_element)
```

Comment:
mv artifact_from_node from cnudie.iter, e.g. to this module. it is not a good match for cnudie.iter, I think